presto-main/src/main/java/com/facebook/presto/sql/analyzer/AggregationAnalyzer.java

```diff
@@ -334,11 +351,21 @@ public Boolean visitWindowFrame(WindowFrame node, Void context)
         @Override
         protected Boolean visitQualifiedNameReference(QualifiedNameReference node, Void context)
         {
-            QualifiedName name = node.getName();
+            return isField(node.getName());
+        }
 
-            List<Field> fields = tupleDescriptor.resolveFields(name);
-            Preconditions.checkState(!fields.isEmpty(), "No fields for name '%s'", name);
-            Preconditions.checkState(fields.size() <= 1, "Found more than one field for name '%s': %s", name, fields);
+        @Override
+        protected Boolean visitDereferenceExpression(DereferenceExpression node, Void context)
```

Comment:
This doesn't handle dereferences which are row field references correctly. For example, try this test case:

```
 assertQuery("SELECT a.col1[1].col0, SUM(a.col0), SUM(a.col1[1].col1), SUM(a.col1[2].col0), SUM(a.col2.col1) FROM " +
                        "(VALUES " +
                        "(ROW(test_row(1.0, ARRAY[test_row(31, 4.5), test_row(12, 4.2)], test_row(3, 4.0)))), " +
                        "(ROW(test_row(3.1, ARRAY[test_row(41, 3.1), test_row(32, 4.2)], test_row(6, 6.0)))), " +
                        "(ROW(test_row(2.2, ARRAY[test_row(31, 4.2), test_row(22, 4.2)], test_row(5, 4.0))))) t(a) " +
                        "GROUP BY a.col1[1]",
                "SELECT * FROM VALUES (31, 3.2, 8.7, 34, 8.0), (41, 3.1, 3.1, 32, 6.0)");
```